File: 2868-continuous-subarrays/2868-continuous-subarrays.java
import java.util.*;

class Solution {
    public long continuousSubarrays(int[] nums) {
        long ans = 0;
        int l = 0, r = 0, n = nums.length;
        Map<Integer, Integer> map = new HashMap<>(); // value -> frequency

        while (r < n) {
            map.put(nums[r], map.getOrDefault(nums[r], 0) + 1);

            int maxi = Integer.MIN_VALUE;
            int mini = Integer.MAX_VALUE;

            for (int key : map.keySet()) {
                maxi = Math.max(maxi, key);
                mini = Math.min(mini, key);
            }

            while (maxi - mini > 2) {
                map.put(nums[l], map.get(nums[l]) - 1);
                if (map.get(nums[l]) == 0) {
                    map.remove(nums[l]);
                }
                l++;

                maxi = Integer.MIN_VALUE;
                mini = Integer.MAX_VALUE;

                for (int key : map.keySet()) {
                    maxi = Math.max(maxi, key);
                    mini = Math.min(mini, key);
                }
            }

            ans += (r - l + 1);
            r++;
        }

        return ans;
    }
}

File: 2868-continuous-subarrays/2868-continuous-subarrays.cs
using System;
using System.Collections.Generic;

public class Solution {
    public long ContinuousSubarrays(int[] nums) {
        long ans = 0;
        int l = 0, r = 0, n = nums.Length;
        Dictionary<int, int> freq = new Dictionary<int, int>();

        while (r < n) {
            if (!freq.ContainsKey(nums[r])) freq[nums[r]] = 0;
            freq[nums[r]]++;

            int maxi = int.MinValue;
            int mini = int.MaxValue;

            foreach (var key in freq.Keys) {
                maxi = Math.Max(maxi, key);
                mini = Math.Min(mini, key);
            }

            while (maxi - mini > 2) {
                freq[nums[l]]--;
                if (freq[nums[l]] == 0) {
                    freq.Remove(nums[l]);
                }
                l++;

                maxi = int.MinValue;
                mini = int.MaxValue;

                foreach (var key in freq.Keys) {
                    maxi = Math.Max(maxi, key);
                    mini = Math.Min(mini, key);
                }
            }

            ans += (r - l + 1);
            r++;
        }

        return ans;
    }
}
